File: NetBet/ApiControllers/SeasonController.fs
namespace NetBet.Controllers

open System
open System.Collections.Generic
open System.Linq
open Microsoft.AspNetCore.Mvc
open DbTypes
open DtoTypes

[<Route("api/[controller]/[action]")>]
[<ApiController>]
type SeasonController () =
    inherit ControllerBase()
    
    [<HttpGet>]
    member __.Get() =
        let seasons = SeasonService.getAllSeasons()
        ActionResult<Season[]>(seasons)

    [<HttpGet("{id}")>]
    member __.GetById(id:int) =
        let season = SeasonService.getSeasonByID(id)
        ActionResult<Season>(season)

    [<HttpGet>]
    member __.GetEmpty() =
        let emptySeason = 
            { ID            = 0
              Name          = ""
              StartTime     = DateTime.Now
              EndTime       = DateTime.Now
              StartingCash  = 0
              MinimumCash   = 0
              MaxParlaySize = 0 }
        let empty = 
            { Season = emptySeason
              Players = [| |] }
        ActionResult<SeasonWithPlayers>(empty)

    [<HttpPost>]
    member __.Create([<FromBody>] sp: SeasonWithPlayers) =
        SeasonService.createSeasonWithPlayers sp |> ignore
        
    [<HttpDelete("{id}")>]
    member __.Delete(id:int) =
        SeasonService.deleteSeason id

File: NetBet.WebApp/ApiControllers/SeasonController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;
using static DbTypes;
using static DtoTypes;

namespace NetBet.WebApp.ApiControllers
{
    [Route("api/[controller]/[action]")]
    [ApiController]
    public class SeasonController : ControllerBase
    {
        [HttpGet]
        public Season[] GetAll()
        { return SeasonService.getAllSeasons(); }
        
        [HttpGet("{id}")]
        public Season GetById(int id)
        { return SeasonService.getSeasonByID(id);}

        [HttpGet("{id}")]
        public SeasonWithPlayers GetSeasonWithPlayers(int id)
        { return SeasonService.getSeasonWithPlayers(id); }

        [HttpGet("{id}")]
        public FullSeason GetFullSeason(int id)
        { return SeasonService.getFullSeason(id); }
        
        [HttpPost]
        public void Edit([FromBody] EditSeasonDto s)
        { SeasonService.createSeasonWithPlayers(s); }
        
        [HttpDelete("{id}")]
        public void Delete(int id)
        { SeasonService.deleteSeason(id); }
    }
}
